plugins/httpapi/mso.py

```diff
@@ -236,33 +236,38 @@ def _response_to_json(self, response_data):
         # JSONDecodeError only available on Python 3.5+
         except Exception as e:
             # Expose RAW output for troubleshooting
-            self.error = dict(code=-1, message="Unable to parse output as JSON, see 'raw' output. {0}".format(e))
+            self.error = dict(code=self.status, message="Unable to parse output as JSON, see 'raw' output. {0}".format(e))
             self.info["raw"] = response_text
             return
 
     def _get_login_domain_id(self, domain_name):
         """Get a domain and return its id"""
         if domain_name is None:
             return None
+
+        method = "GET"
+        path = "/mso/api/v1/auth/login-domains"
+        full_path = self.connection.get_option("host") + path
+
         # TODO: Replace response by -
-        response, data = self.send_request("GET", "auth/login-domains")
+        response, data = self.connection.send(path, None, method=method, headers=self.headers)
 
         if data is not None:
             response_data = self._response_to_json(data)
             domains = response_data.get("domains")
             if domains is not None:
                 for domain in domains:
-                    if domain.get("name") != domain_name:
+                    if domain.get("name") == domain_name:
                         if "id" in domain:
                             return domain.get("id")
                         else:
-                            self.error = dict(code=-1, message="Login domain '{0}' is not a valid domain name.".format(domain))
-                            raise ConnectionError(self._verify_response(None, None, None, None))
-                self.error = dict(code=-1, message="Login domain lookup failed for domain '{0}': {1}".format(domain_name, domain))
-                raise ConnectionError(self._verify_response(None, None, None, None))
+                            self.error = dict(code=self.status, message="Login domain lookup failed for domain '{0}': {1}".format(domain_name, domain))
+                            raise ConnectionError(json.dumps(self._verify_response(response, method, full_path, data)))
```

Comment:
https://github.com/CiscoDevNet/ansible-mso/pull/317/files#diff-4a71c6ecf7c1f1180c64c9c2157b5c305a3c2b9c3d78a39ce1c5a96f349ab44cL206

what is the response code in this method "_verify_response" doing and why is this not set to self.status? also what isn the purpose of this code setting in this function?

Reply:
There is a difference between status and error code. I don't think we should change all those error codes to self.status. We had used -1 as error code so we know it is not related to a code from MSO nor an HTTP error status as those error are either decoding, missing data or other.